### ISSUE TITLE
Parse Credentials from URL for Proxies that use auth

### ISSUE THREAD

**Issue by cjames23** [chadac/poetry-plugin-pypi-proxy#6]
When using the plugin against a proxy Code Artifact that utilizes credentials for auth the plugin throws exception for parsing the url. 

```
AttributeError

  'NoneType' object has no attribute 'groups'

  at /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/urllib3/util/url.py:370 in parse_url
      366│ 
      367│         if authority:
      368│             auth, _, host_port = authority.rpartition("@")
      369│             auth = auth or None
    → 370│             host, port = _HOST_PORT_RE.match(host_port).groups()
      371│             if auth and normalize_uri:
      372│                 auth = _encode_invalid_chars(auth, USERINFO_CHARS)
      373│             if port == "":
      374│                 port = None

The following error occurred when trying to handle this error:


  Stack trace:

  2  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/requests/models.py:382 in prepare_url
      380│         # Support for unicode domain names and paths.
      381│         try:
    → 382│             scheme, auth, host, port, path, query, fragment = parse_url(url)
      383│         except LocationParseError as e:
      384│             raise InvalidURL(*e.args)

  1  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/urllib3/util/url.py:394 in parse_url
      392│ 
      393│     except (ValueError, AttributeError):
    → 394│         return six.raise_from(LocationParseError(source_url), None)
      395│ 
      396│     # For the sake of backwards compatibility we put empty

  LocationParseError

  Failed to parse: https://aws/@{}:amazon-{}.d.codeartifact.us-west-2.amazonaws.com/pypi/{}/

  at <string>:3 in raise_from
        1│ 

The following error occurred when trying to handle this error:


  Stack trace:

  16  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cleo/application.py:329 in run
       327│ 
       328│             try:
     → 329│                 exit_code = self._run(io)
       330│             except Exception as e:
       331│                 if not self._catch_exceptions:

  15  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/poetry/console/application.py:185 in _run
       183│         self._load_plugins(io)
       184│ 
     → 185│         exit_code: int = super()._run(io)
       186│         return exit_code
       187│ 

  14  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cleo/application.py:423 in _run
       421│             io.input.set_stream(stream)
       422│ 
     → 423│         exit_code = self._run_command(command, io)
       424│         self._running_command = None
       425│ 

  13  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cleo/application.py:465 in _run_command
       463│ 
       464│         if error is not None:
     → 465│             raise error
       466│ 
       467│         return event.exit_code

  12  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cleo/application.py:449 in _run_command
       447│ 
       448│             if event.command_should_run():
     → 449│                 exit_code = command.run(io)
       450│             else:
       451│                 exit_code = ConsoleCommandEvent.RETURN_CODE_DISABLED

  11  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cleo/commands/base_command.py:119 in run
       117│         io.input.validate()
       118│ 
     → 119│         status_code = self.execute(io)
       120│ 
       121│         if status_code is None:

  10  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cleo/commands/command.py:83 in execute
        81│ 
        82│         try:
     →  83│             return self.handle()
        84│         except KeyboardInterrupt:
        85│             return 1

   9  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/poetry/console/commands/publish.py:82 in handle
        80│         )
        81│ 
     →  82│         publisher.publish(
        83│             self.option("repository"),
        84│             self.option("username"),

   8  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/poetry/publishing/publisher.py:86 in publish
        84│         )
        85│ 
     →  86│         self._uploader.upload(
        87│             url,
        88│             cert=resolved_cert,

   7  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/poetry/publishing/uploader.py:125 in upload
       123│ 
       124│         try:
     → 125│             self._upload(session, url, dry_run, skip_existing)
       126│         finally:
       127│             session.close()

   6  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/poetry/publishing/uploader.py:216 in _upload
       214│             # TODO: Check existence
       215│ 
     → 216│             self._upload_file(session, url, file, dry_run, skip_existing)
       217│ 
       218│     def _upload_file(

   5  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/poetry/publishing/uploader.py:256 in _upload_file
       254│             try:
       255│                 if not dry_run:
     → 256│                     resp = session.post(
       257│                         url,
       258│                         data=monitor,

   4  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/requests/sessions.py:590 in post
       588│         """
       589│ 
     → 590│         return self.request('POST', url, data=data, json=json, **kwargs)
       591│ 
       592│     def put(self, url, data=None, **kwargs):

   3  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/requests/sessions.py:528 in request
       526│             hooks=hooks,
       527│         )
     → 528│         prep = self.prepare_request(req)
       529│ 
       530│         proxies = proxies or {}

   2  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/requests/sessions.py:456 in prepare_request
       454│ 
       455│         p = PreparedRequest()
     → 456│         p.prepare(
       457│             method=request.method.upper(),
       458│             url=request.url,

   1  /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/requests/models.py:316 in prepare
       314│ 
       315│         self.prepare_method(method)
     → 316│         self.prepare_url(url, params)
       317│         self.prepare_headers(headers)
       318│         self.prepare_cookies(cookies)

  InvalidURL

  Failed to parse: https://aws/@{}:amazon-{}.d.codeartifact.us-west-2.amazonaws.com/pypi/{}/

  at /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/requests/models.py:384 in prepare_url
      380│         # Support for unicode domain names and paths.
      381│         try:
      382│             scheme, auth, host, port, path, query, fragment = parse_url(url)
      383│         except LocationParseError as e:
    → 384│             raise InvalidURL(*e.args)
      385│ 
      386│         if not scheme:
      387│             error = ("Invalid URL {0!r}: No schema supplied. Perhaps you meant http://{0}?")
      388│             error = error.format(to_native_string(url, 'utf8'))
```

Stack traces have sensitive information removed.

**Comment by cjames23** [comment]
Should have a PR ready tomorrow. Just need to fix some typical mypy being mypy stuff.

**Comment by chadac** [comment]
Resolved by #9
